platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -0,0 +1,290 @@
+		// of the Candidate Roster for adoption. Once the metric is at 100%, this shows
+		// that the network can be restarted in order to adopt the Candidate Roster.
+	}
+	*/
+            }
+
+            if(
+the State
+has no
+Active Roster){
+
+// This should never happen
+throwFatalErrorAndShutdown();
+}
+
+// Check if the Active Roster is TSS-enabled
+        if(
+Active Roster
+is not
+TSS-enabled){
+  /*
+  // This block will be uncommented once the TSS protocol is implemented
+	if (isGenesis) {
+		rejectAnythingButTSSMessages = true;
+		// Note that a TSS upgrade of an existing network
+		// shouldn't disable non-TSS messages processing.	
+	}
+	*/
+
+        // In Genesis, we've just disabled processing anything but TSS. So a new
+        // network will eventually become TSS-enabled and only then will start
+        // processing non-TSS events.
+        // An existing, non-genesis network will continue to operate as before,
+        // and if a Candidate Roster was present in the state, we enabled processing TSS messages
+        // above. So it will become TSS-enabled eventually, too, indicating
+        // the readiness for a TSS upgrade via the metric mentioned above.
+        // Note that until a Candidate Roster is installed in the state, the existing network
+        // will simply continue to operate as usual with its regular RSA keys.
+        }
+
+// At this point the Active Roster in the state is what we'll be using as a roster.
+// If an incomplete Candidate Roster exists, we've started exchanging TSS messages above.
+// If it's genesis, we've disabled processing anything but TSS messages above.
+// If Active Roster is already TSS-enabled (a mature network running after the TSS
+// upgrade), then the node will be able to sign using TSS.
+// If Active Roster isn't TSS-enabled yet, then the node will sign using its RSA key only,
+// just as it does today.
+```
+
+### Bootstrapping Genesis for a brand new network
+
+This proposal already creates a new interface for the services to install a Candidate Roster—via a Platform API, or
+similar—which ultimately stores the Candidate Roster in a designated location in the State.
+The Startup procedure described above consumes the Candidate Roster from that location and starts its processing or
+adoption based on its readiness.
+
+The `config.txt` will only ever be used literally once—when a brand new network goes through its genesis.
+The `config.txt` will never be used again in the network life cycle.
+
+Given that:
+
+1. we need a special mechanism for supplying a genesis roster to a brand new network, and
+2. this mechanism is only ever used once during genesis, and
+3. we already have a support for `config.txt` in the platform, and
+4. we already propose a public interface for Services to install Candidate Rosters in existing networks
+
+there should be absolutely no need to introduce any additional mechanisms or interfaces between Platform and Services to
+supply a genesis roster.
```

Comment:
this is false, see my last comment.    If we don't do inversion of control then we have to manage data that services needs in config.txt that is not stored in the roster.